packages/cli/src/ceramic-logger-plugins.ts

```diff
@@ -86,28 +113,125 @@ export class LogToFiles {
+     * @param filePath Path to file that is blocked
+     */
+    private static async _writeBlockedWarning (
+        blockedFiles: BlockedFiles,
+        basePath: string,
+        filePath: string
+    ): Promise<void> {
+        const loggerFilePath = basePath + 'logger.log'
+        const message = `Stream blocked for file: ${filePath}\n`
+        await LogToFiles._writeStream(blockedFiles, loggerFilePath, message, 'a')
+    }
+
+    /**
+      * Writes `message` to a file and returns whether it is blocked or not
+      * @notice If the file is currently being written to by an open stream or if
+      * its buffer is full, the call to this method will return without writing
+      * @param filePath Full path of file to write to
+      * @param message Message to write to `filePath`
+      * @param writeFlag Specifies writing method (e.g. "a" for append, "w" for overwrite)
+      * @returns True if the stream is blocked and false otherwise
+      */
+    private static async _writeStream (
+        blockedFiles: BlockedFiles,
+        filePath: string,
+        message: string,
+        writeFlag: string
+    ): Promise<WriteResult> {
+        if (blockedFiles[filePath]) {
+            console.warn(`Stream busy for ${filePath}. Some logs may be dropped.`)
+            return { blocked: blockedFiles[filePath], filePath }
+        }
+
+        blockedFiles[filePath] = true
+
+        const stream = fs.createWriteStream(filePath, { flags: writeFlag })
+        stream.on('error', (err) => {
+            console.warn(err)
+            return
+        })
+        stream.on('drain', () => {
+            blockedFiles[filePath] = false
+            return
+        })
+        stream.on('finish', () => {
+            blockedFiles[filePath] = false
+            return
+        })
+        blockedFiles[filePath] = !stream.write(util.format(message) + '\n', () => {
+            stream.end()
+            return
+        })
+        return { blocked: blockedFiles[filePath], filePath }
+    }
+
+    /**
+     * Returns true if the file should be rotated, false otherwise
      * @param filePath Full path of file
+     * @returns If the file should be rotated
      */
-    private static async _isExpired (filePath: string): Promise<boolean> {
+    private static async _shouldRotate (filePath: string): Promise<boolean> {
+        let fileStats = null
+
         try {
-            const { birthtime } = await fsPromises.stat(filePath)
-            const minutesSinceBirth = (Date.now() - birthtime.getTime()) / (1000 * 60)
-            return minutesSinceBirth >= LogToFiles.MINUTES_TO_EXPIRATION
+            fileStats = await fsPromises.stat(filePath)
         } catch (err) {
-            if (err.code == 'ENOENT') return false
-            else console.error(err)
+            if (err.code == 'ENOENT') { // File not found
+                return false
+            }
+            console.error(err)
+            return true
+        }
+
+        if (!fileStats) {
```

Comment:
Looks like `LogToFiles._isExpired` and `LogToFiles._isMaxSize` don't need to be async so this logic could be reduced to `return !fileStats || LogToFiles._isExpired(fileStats) || LogToFiles._isMaxSize(fileStats)`

Reply:
oh thank you! forgot to clean that up